node/src/p2p/header_session.rs

```diff
@@ -1,52 +1,51 @@
 use celestia_proto::p2p::pb::HeaderRequest;
 use celestia_types::ExtendedHeader;
 use tokio::sync::{mpsc, oneshot};
-use tracing::debug;
+use tracing::{debug, warn};
 
 use crate::executor::spawn;
 use crate::p2p::header_ex::utils::HeaderRequestExt;
 use crate::p2p::{HeaderExError, P2pCmd, P2pError};
+use crate::store::utils::{HeaderRanges, HeaderRangesIterator, RangeLengthExt};
 
 const MAX_AMOUNT_PER_REQ: u64 = 64;
-const MAX_CONCURRENT_REQS: usize = 8;
+const MAX_CONCURRENT_REQS: usize = 1;
 
 type Result<T, E = P2pError> = std::result::Result<T, E>;
 
 pub(crate) struct HeaderSession {
-    next_height: u64,
-    remaining_amount: u64,
+    ranges_iter: HeaderRangesIterator,
+    ranges_count: usize,
     cmd_tx: mpsc::Sender<P2pCmd>,
     response_tx: mpsc::Sender<(u64, u64, Result<Vec<ExtendedHeader>>)>,
     response_rx: mpsc::Receiver<(u64, u64, Result<Vec<ExtendedHeader>>)>,
     ongoing: usize,
 }
 
 impl HeaderSession {
-    pub(crate) fn new(from_height: u64, amount: u64, cmd_tx: mpsc::Sender<P2pCmd>) -> Result<Self> {
-        if from_height < 1 || amount < 1 {
-            return Err(P2pError::HeaderEx(HeaderExError::InvalidRequest));
-        }
+    pub(crate) fn new(ranges: HeaderRanges, cmd_tx: mpsc::Sender<P2pCmd>) -> Result<Self> {
```

Comment:
Add a bit of documentation on how `HeaderRanges` are expected to be constructed. An example of an input and a possible output (example of output: `[1, 2, 3, 4], [7, 8]`)

Reply:
Added documentation with links to relevant functions and examples of parameters and responses.